prow/github/report/report.go

```diff
@@ -190,7 +190,18 @@ func ReportComment(ctx context.Context, ghc GitHubClient, reportTemplate *templa
 			return fmt.Errorf("error deleting comment: %w", err)
 		}
 	}
-	if len(entries) > 0 || mustCreate {
+
+	// If there are any aborted pjs for this ref we don't want to report that all tests passed.
+	// This could be due to a push while pjs are running.
+	aborted := false
+	for _, pj := range validPjs {
+		if pj.Status.State == v1.AbortedState {
```

Comment:
I believe that this is okay due to
https://github.com/kubernetes/test-infra/pull/27077/files#diff-aded99920ab3596285f32941b92ca495de84e490df63313d717b8389ec10ab23L170-R172
my assumption is this means they will all be based on the same SHA, and we won't be looking at older aborted jobs.

Reply:
When there is a new patch pushed on the PR, all the current jobs should be aborted, which is guarded by https://github.com/kubernetes/test-infra/blob/8fd98f95dc8f7cd34bfccc19277da88a78c598f7/prow/crier/reporters/github/reporter.go#L178. I think what you ran into was a case where plank was not fast enough killing previous jobs.

I don't think relying on aborted state is the most reliable way of doing this, as jobs could be aborted with other reasons and some of them might worth reporting. A better fix imo, is at https://github.com/kubernetes/test-infra/blob/8fd98f95dc8f7cd34bfccc19277da88a78c598f7/prow/crier/reporters/github/reporter.go#L178, make the `pjsToReport` function return an empty slice when there is any prow job triggered from a newer patch

Reply:
> my assumption is this means they will all be based on the same SHA

That is incorrect, because the jobs we abort on push are guaranteed to not be for the latest sha.

> make the pjsToReport function return an empty slice when there is any prow job triggered from a newer patch

Don't we already do that due to the `pjob.Complete()` check if we actually observe a newer one?

Reply:
> > make the pjsToReport function return an empty slice when there is any prow job triggered from a newer patch
> 
> Don't we already do that due to the `pjob.Complete()` check if we actually observe a newer one?

If you mean here, I don't think it checks previous runs https://github.com/kubernetes/test-infra/blob/7ff165159229e3418880928e3fcda5bc93aac393/prow/apis/prowjobs/v1/types.go#L945-L948

Reply:
We select on org/repo/pr from the cache, then return nothin if we find any prowjob that is incomplete. This implies that if there is a prowjob for a newer sha of the repo that hasn't finished, we won't be reporting anything or not?

Reply:
oh I had misunderstood you. Initially I thought the list prowjobs function would filter based on the head SHA, but apparently it's not the case. So you're right that if there is any new prowjob triggered from a newer commit, nothing will be returned here. So it's still unclear to me why https://github.com/kubernetes/test-infra/issues/27072 had happened, and I'm not convinced that this PR would fix the problem

Reply:
> That is incorrect, because the jobs we abort on push are guaranteed to not be for the latest sha.

@alvaroaleman they aren't from the latest sha, but to get in this state don't they need to be from the sha that was just aborted? I think when we are in this scenario we have a list of jobs (from the same sha) of which some may have succeeded, but others have been aborted. Due to that, the current implementation comments that all of the tests have passed.

Note also that when some of the jobs have already failed prior to the new commit being pushed we end up listing those failed jobs in the comment, but **none** of the aborted jobs.

I think that if this function was _only_ being called with the latest sha we would never see this bug as the jobs wouldn't be marked as `complete` since they are still running.

Reply:
In general, it makes sense to me to definitely not comment that "All tests have passed" when we have aborted tests in the sha we are referencing.

Reply:
You are right, I missed the fact that we will only include the newest prowjob of a given name in the list we pass to `ReportComment`, so this won't be blocking reporting altogether: https://github.com/kubernetes/test-infra/blob/814aefde6ec0b268a9af398cc5d292141d048d3b/prow/crier/reporters/github/reporter.go#L182-L184

The only situation I can think of where this would do something unexpected now is that it won't report if someone manually aborts a ProwJob through Deck. Then again, I can not think of a good reason to do that for a presubmit and that happening is probably an edge case.

> Note also that when some of the jobs have already failed prior to the new commit being pushed we end up listing those failed jobs in the comment, but none of the aborted jobs.

I think that is simply because we only list failed jobs, not jobs in any other state:
https://github.com/kubernetes/test-infra/blob/814aefde6ec0b268a9af398cc5d292141d048d3b/prow/github/report/report.go#L284